src/util/std_types.h

```diff
@@ -1671,4 +1671,66 @@ inline complex_typet &to_complex_type(typet &type)
   return static_cast<complex_typet &>(type);
 }
 
+class function_typet:public typet
+{
+public:
+  function_typet():typet(ID_function)
+  {
+    subtypes().resize(2);
+  }
+
+  class variablet:public irept
```

Comment:
Is this different from `symbol_exprt` in some way?

Reply:
I would say yes. In particular, the identifier need not be set (say you want to have f:N->N). Will add comment to that extend.

Reply:
Is "variable" the most appropriate name? Isn't it a parameter?

Reply:
No, parameter is programming language terminology.

Reply:
Indeed "parameter" seems to be the only non-option. Having read the wikipedia page, I'd go with `input_variablet` or `argumentt`. The name `variablet` would make sense if declared at global scope, i.e., outside `function_typet`.

Reply:
Arguments are (as in programs) the values that you give to a variable, but I see the risk that these are confused with program variables. The fact that variablet is INSIDE mathematical_function_typet should help?

Reply:
Hmhm, will switch to "domain".